submissions/scripts/export_lims_metadata.py

```diff
@@ -0,0 +1,190 @@
+
+        return df
+    else:
+        raise Exception(f'Error:\n Empty response for: {body}')
+
+
+def get_lims_additional_sequencing_metadata(barcodes: list):
+    body = json.dumps([{"anyBarcode": b} for b in barcodes if not str(b).endswith("_exp")])
+    response = lims_api_request(body, '/api/v1/sfs-specimens/sequencing-submission-metadata-for-or-ha')
+
+    if response.content is not None:
+        df = pd.json_normalize(json.loads(response.content), max_level=3)
+
+        column_map = {
+            'subject.participantId': 'participant_id',
+            'subject.firstName': 'first_name',
+            'subject.lastName': 'last_name',
+            'subject.birthdate': 'birthdate',
+            'subject.sex': 'sex',
+            'subject.address.street': 'street',
+            'subject.address.city': 'city',
+            'subject.address.state': 'state',
+            'subject.address.zipCode': 'zip',
+            'subject.phone': 'phone',
+            'specimen.ids.sampleId': 'sfs_sample_identifier',
+            'specimen.ids.sampleBarcode': 'sfs_sample_barcode',
+            'specimen.ids.collectionBarcode': 'sfs_collection_barcode',
+            'specimenCollectionSite': 'specimen_collection_site'
+        }
+        df.rename(columns=column_map, inplace=True)
+
+        return df
+    else:
+        raise Exception(f'Error:\n Empty response for: {body}')
+
+
+def add_lims_metadata(input_df:pd.DataFrame) -> pd.DataFrame:
+    barcodes = input_df['sfs_sample_barcode'].to_list()
+    result = get_lims_sequencing_metadata(barcodes)
+    print(f"Received {len(result)} metadata records from LIMS.")
+
+    output_df = pd.merge(input_df, result, on=['sfs_sample_barcode'])
+    remaining_records = input_df[(~input_df.nwgc_id.isin(output_df.nwgc_id))]
+    remaining_records_joined = pd.merge(remaining_records, result, left_on='sfs_sample_barcode', right_on='sfs_collection_barcode')
+
+    output_df = pd.concat([output_df, remaining_records_joined], ignore_index=True).sort_values(by=['nwgc_id'])
+
+    OUTPUT_COLS = [
+        'nwgc_id',
+        'sfs_sample_identifier',
+        'sfs_sample_barcode',
+        'sfs_collection_barcode',
+        'sfs_identifier_for_doh_reporting',
+        'collection_date',
+        'swab_type',
+        'source',
+        'puma',
+        'county',
+        'state',
+        'baseline_surveillance',
+        'error_message'
+    ]
+
+    return output_df[OUTPUT_COLS]
+
+
+def add_lims_additional_metadata(input_df:pd.DataFrame) -> pd.DataFrame:
+    barcodes = input_df[input_df['state'].str.lower() == 'oregon']['sfs_collection_barcode'].to_list()
+    result = get_lims_additional_sequencing_metadata(barcodes)
+    print(f"Received {len(result)} additional metadata (Oregon) records from LIMS.")
+
+    output_df = pd.merge(input_df, result, on=['sfs_collection_barcode'])
+    #remaining_records = input_df[(~input_df.nwgc_id.isin(output_df.nwgc_id))]
```

Comment:
Are these commented lines used on occasion?

Reply:
For now, the only OR samples are from Cascadia, and it should always be matching on collection barcode for that project, so the follow-up join on sample barcodes isn't necessary.